src/main/java/org/folio/linked/data/service/impl/tenant/TenantScopedExecutionService.java

```diff
@@ -17,35 +18,45 @@
 import org.folio.spring.tools.context.ExecutionContextBuilder;
 import org.springframework.context.annotation.Profile;
 import org.springframework.messaging.MessageHeaders;
+import org.springframework.retry.support.RetryTemplate;
 import org.springframework.scheduling.annotation.Async;
 import org.springframework.stereotype.Service;
 
 @Service
 @Profile(FOLIO_PROFILE)
 @RequiredArgsConstructor
 public class TenantScopedExecutionService {
+  private final RetryTemplate retryTemplate;
   private final ExecutionContextBuilder contextBuilder;
 
   @SneakyThrows
-  public <T> T executeTenantScoped(String tenantId, Callable<T> job) {
+  public <T> T execute(String tenantId, Callable<T> job) {
     try (var fex = new FolioExecutionContextSetter(dbOnlyContext(tenantId))) {
       return job.call();
     }
   }
 
-  public void executeTenantScoped(String tenantId, Runnable job) {
+  public void execute(String tenantId, Runnable job) {
     try (var fex = new FolioExecutionContextSetter(dbOnlyContext(tenantId))) {
       job.run();
     }
   }
 
   @Async
-  public void executeAsyncTenantScoped(Headers headers, Runnable job) {
+  public void executeAsyncWithRetry(Headers headers, Runnable job, Consumer<Throwable> failureHandler) {
     try (var fex = new FolioExecutionContextSetter(kafkaFolioExecutionContext(headers))) {
-      job.run();
+      retryTemplate.execute(
+        context -> runAndReturn(job, true),
+        context -> runAndReturn(() -> failureHandler.accept(context.getLastThrowable()), false)
+      );
     }
   }
 
+  private boolean runAndReturn(Runnable job, boolean returnValue) {
```

Comment:
minor
sugest to do 2 simple separate methods without boolean in parameters
```
private boolean runJob(Runnable job) {
  job.run();
  return true;
}
private boolean handeError(Consumer<Throwable> failureHandler, Context contex) {
  failureHandler.accept(context.getLastThrowable())
  return fasle;
```

Reply:
Done.